src/axom/quest/detail/MeshTester_detail.hpp

```diff
@@ -168,9 +168,9 @@ void CandidateFinderBase<ExecSpace, FloatType>::initialize()
   // if the Triangle is degenerate.
   mint::for_all_cells<ExecSpace, mint::xargs::coords>(
     m_surfaceMesh,
-    AXOM_LAMBDA(IndexType cellIdx,
-                numerics::Matrix<double> & coords,
-                const IndexType* nodeIds) {
+    [&](IndexType cellIdx,
```

Comment:
This was the bugfix:
* `AXOM_LAMBDA` adds a by-value lambda capture (`[=]`)
* Since we modify data in this kernel, we need a by-reference lambda capture (`[&]`)
* I am not sure why gcc@8.1 gave an error (related to modifying const data) but not other compilers

There are two other similar changes in this file.

Reply:
That's curious, what was the error we were getting? I think longer-term we should aim to fix whatever's breaking in `ArrayView::operator[]`, but this workaround is fine for now.

Reply:
I confirmed that the problem is a bug in `gcc@8.1` that is fixed in `gcc@8.3`.

As such, I removed the `AXOM_MUTABLE_LAMBDA` macro from `Macros.hpp` in favor of a local macro in `MeshTester_detail.hpp` that uses by-reference lambda-capture for gcc@8.1 and `AXOM_LAMBDA` otherwise.

I tested this against all our toss3 and blueos configurations as well as against `gcc@8.1` and `gcc@8.3`, with and without RAJA + Umpire.